client/state/selectors/get-is-eligible-for-intro-offer.ts

```diff
@@ -0,0 +1,25 @@
+import { get } from 'lodash';
+import type { AppState } from 'calypso/types';
+
+/**
+ * @param  {object}  state       Global state tree
+ * @param  {number}  productId   The productId to check for an intro offer
+ * @param  {number?}  siteId      The ID of the site we're querying
+ * @returns {boolean}        The raw price of intro offer, if available. null otherwise.
+ */
+export default function getIsEligibleForIntroOffer(
+	state: AppState,
+	productId: number,
+	siteId: number | 'none'
+): boolean {
+	const ineligibleReason = get(
```

Comment:
This could probably be written without lodash, maybe like this?

```js
const ineligibleReason = state.sites.introOffers?.[ items ]?.[ siteId ]?.[ productId ]?.ineligibleReason ?? null;
```